app/ts/components/SendEthPage.tsx

```diff
@@ -46,30 +48,168 @@ const Page = ({ children }: { children: ComponentChildren }) => {
+		txn.value.status === 'idle' && txn.value.compose()
+	}, [])
+
+	return <form onSubmit={handleSubmit}>{children}</form>
+}
+
+const isEditInputAllowed = (store: TransactionStore): store is TransactionComposing | TransactionFailed =>
+	store.status === 'composing' || store.status === 'failed'
+
+const SendAmountField = () => {
+	const txn = sendTransactionStore
+
+	function handleChange(amount: string) {
+		if (!isEditInputAllowed(txn.value)) return;
+		txn.value.data = { ...txn.value.data, amount }
+		if (txn.value.status === 'failed') txn.value.reset()
+	}
+
+	const inputValue = txn.value.status === 'idle' ? '' : txn.value.data.amount
+	const disabled = !isEditInputAllowed(txn.value)
+
+	return <AmountField value={inputValue} onChange={handleChange} label='Amount' name='amount' disabled={disabled} />
+}
+
+const SendAddressField = () => {
+	const txn = sendTransactionStore
+
+	function handleChange(to: string) {
+		if (!isEditInputAllowed(txn.value)) return;
+		txn.value.data = { ...txn.value.data, to }
+		if (txn.value.status === 'failed') txn.value.reset()
+	}
+
+	const inputValue = txn.value.status === 'idle' ? '' : txn.value.data.to
+	const disabled = !isEditInputAllowed(txn.value)
+
+	return <AddressField value={inputValue} onChange={handleChange} label='Address' name='to' disabled={disabled} />
+}
+
+const SendGuide = () => {
+	const txn = sendTransactionStore
+
+	switch (txn.value.status) {
+		case 'composing':
+			return (
+				<div class='p-4 text-center xl:text-left'>
+					<div class='mb-2'>What happens when I click send?</div>
+					<div class='leading-tight text-white/50 text-sm'>This app will forward your request to the wallet you chose to connect with. The connected wallet handles signing and submitting your requst to the chain.</div>
+				</div>
+			)
+		case 'signing':
+			return (
+				<div class='p-4 text-center xl:text-left'>
+					<div class='mb-2 font-bold'>Awaiting wallet confirmation...</div>
+					<div class='leading-tight text-white/50 text-sm'>At this point, your connected wallet will need action to proceed with this transaction. Carefully check the information before accepting the wallet confirmation.</div>
+				</div>
+			)
+		case 'signed':
+			return (
+				<div class='p-4 text-center xl:text-left'>
+					<div class='mb-2 font-bold'>Request Successfully Sent!</div>
+					<div class='leading-tight text-white/50 text-sm'>Your transaction is awaiting confirmation from the chain. You may click on the View Transaction button to check it's status.</div>
+				</div>
+			)
+		case 'failed':
+			return (
+				<div class='p-4 text-center xl:text-left'>
+					<div class='mb-2 font-bold'>Wallet returned an error!</div>
+					<div class='leading-tight xl:border-l-4 xl:border-l-white/30 px-3 py-1 flex items-center justify-center xl:justify-start text-white/50 text-sm'>{txn.value.error.message}</div>
+					<div class='text-sm mt-2'>Check that your inputs are correct and click Send again.</div>
+				</div>
+			)
+		default:
```

Comment:
list the other states here instead of default, so when new states are added, the typescript would request you to handle those cases too

Reply:
made the adjustments here f51fe8e96384df911564865229bee5cc0fbc7486